packages/p2p/src/components/my-profile/my-profile.jsx

```diff
@@ -1,70 +1,346 @@
+            getAdvertiserStats();
+            getAdvertiserInfo();
+        }
+    }, [is_mounted.current]);
+
+    const getAdvertiserAccountStatus = () => {
+        return new Promise(resolve => {
+            requestWS({
+                get_account_status: 1,
+            }).then(response => {
+                if (!response.error) {
+                    const { get_account_status } = response;
+                    const { authentication } = get_account_status;
+                    setHasPoa(authentication.document && authentication.document.status === 'verified');
+                    setHasPoi(authentication.identity && authentication.identity.status === 'verified');
+                } else {
+                    setErrorMessage(response.error);
+                }
+                resolve();
+            });
+        });
+    };
+
+    const getAdvertiserInfo = () => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_info: 1,
+            }).then(response => {
+                if (!response.error) {
+                    const { p2p_advertiser_info } = response;
+                    setAdvertiserInfo(p2p_advertiser_info);
+                    setNickname(p2p_advertiser_info.name);
+                    setContactInfo(p2p_advertiser_info.contact_info);
+                    setDefaultAdvertDescription(p2p_advertiser_info.default_advert_description);
+                    setPaymentInfo(p2p_advertiser_info.payment_info);
+                } else {
+                    setErrorMessage(response.error);
+                }
+                resolve();
+                setIsLoading(false);
+            });
+        });
+    };
+
+    const getAdvertiserStats = () => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_stats: 1,
+                id,
+            }).then(response => {
+                if (!response.error) {
+                    const { p2p_advertiser_stats } = response;
+                    setStats(p2p_advertiser_stats);
+                } else {
+                    setErrorMessage(response.error);
+                }
+                resolve();
+            });
+        });
+    };
+
+    const handleSubmit = values => {
+        setIsLoading(true);
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_update: 1,
+                contact_info: values.contact_info,
+                payment_info: values.payment_info,
+                default_advert_description: values.default_advert_description,
+            }).then(response => {
+                if (!response.error) {
+                    const { p2p_advertiser_update } = response;
+                    setContactInfo(p2p_advertiser_update.contact_info);
+                    setDefaultAdvertDescription(p2p_advertiser_update.default_advert_description);
+                    setPaymentInfo(p2p_advertiser_update.payment_info);
```

Comment:
And here too, check for `is_mounted`.

Reply:
Done. Thank you Aaron! Appreciate your review! :)